Makefile

```diff
@@ -60,10 +60,12 @@ build-slim:
 test: run-unit-tests
 
 .PHONY: lint
+lint: diffnoprefix := $(shell sh -c "git config --get --show-origin diff.noprefix | awk '/file:\.git\/config\s+\S+/ {print \$$2}'")
 lint: build-slim build-test
-	# Note: Make sure that you run `git config diff.noprefix false` in this repo
-	# See this issue for more details: https://github.com/golangci/golangci-lint/issues/948
+	# To workaround: https://github.com/golangci/golangci-lint/issues/948
+	git config --replace-all diff.noprefix false
 	golangci-lint run --enable-all --disable=godox,gochecknoglobals,goerr113 --max-same-issues=0 --max-issues-per-linter=0 --build-tags $(ALL_BUILD_TAGS) --new-from-rev=$$(git merge-base $$(cat .git/resource/base_sha 2>/dev/null || echo "origin/master") HEAD) --modules-download-mode=$(MOD) --timeout=5m --exclude-use-default=false ./...
+	if [ -z "$(diffnoprefix)" ]; then git config --unset diff.noprefix; else git config --replace-all diff.noprefix $(diffnoprefix); fi
```

Comment:
This isn't global, right? I'd be careful with doing laptop-level changes on behalf of users...

Reply:
No, it's only touching `.git/config`. For it being global, you would have to use `--global`. It only touches local config and preserves the value, if you have something set there. 

The only "controversial" part is `--replace-all`, if you have multiple values set for the same key, it will wipe them and just set the effective one, but that seem rare enough case to me to accept it.